File: mobile_ui/quickchat_ui/lib/features/chat/usecase/GetConversations.usecase.dart
import 'package:injectable/injectable.dart';
import 'package:quickchat_ui/core/logic/state_status.dart';
import 'package:quickchat_ui/features/chat/data/bloc/chat_bloc.dart';
import 'package:quickchat_ui/features/chat/data/dto/GetConversations.dto.dart';

@injectable
class GetConversationsUsecase {
  GetConversationsUsecase({required this.chatBloc});
  final ChatsBloc chatBloc;

  void execute({
    required StateStatus stateStatus,
    required GetConversationsDto getConversations,
  }) {
    chatBloc.add(ChatsEvent.getConversations(
      stateStatus: stateStatus,
      getConversations: getConversations,
    ));
  }
}

File: web_ui/app/features/chat/domain/usecases/getConversations.usecase.ts
import { StateStatus } from "@/app/core/types";
import { getConversations } from "../../data/redux/chatSlice";
import { GetConversationsDto } from "../../data/dto/GetConversations.dto";

export class GetConversationsUsecase {
  async execute(
    {
      stateStatus,
      getConversationsDto,
    }: {
      stateStatus: StateStatus;
      getConversationsDto: GetConversationsDto;
    },
    dispatch: any
  ): Promise<void> {
    dispatch(getConversations({ stateStatus, getConversationsDto }));
  }
}
